src/Shared/Utilities/VersionComparer.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CST.OpenSource.Shared
+{
+    public class VersionComparer : IComparer<List<string>>
+    {
+
+        /// <summary>
+        /// Convert a version (string) to a list of version parts.
+        /// Just splits into "numbers" and "non-numbers".
+        /// </summary>
+        /// <param name="version"></param>
+        public static List<string> Parse(string version)
+        {
+            var parts = new List<string>();
+            var inNumber = false;
+            var curStr = "";
+
+            for (int i = 0; i < version.Length; i++)
+            {
+                var ch = version[i];
+                if (Char.IsNumber(ch) ^ inNumber)
+                {
+                    if (curStr != "")
+                    {
+                        parts.Add(curStr);
+                    }
+                    curStr = "";
+                }
+
+                curStr += ch;
+                if (i == version.Length - 1)
+                {
+                    parts.Add(curStr);
+                }
+                inNumber = Char.IsNumber(ch);
+            }
+            return parts;
+        }
+
+        /// <summary>
+        /// Compares two lists of strings, using natural numbers where possible.
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns></returns>
+        public int Compare(List<string>? x, List<string>? y)
```

Comment:
I don't think you need this at all.  See other comment about sorting version strings natively.

If you do want to do it this way. I would strongly recommend just converting the numbers into numbers before listing them to make this way simpler.

Reply:
```
for (int i=0; i<minLength; i++)
{
    if (int.TryParse(x[i], out int xInt) &&
        int.TryParse(y[i], out int yInt))
    {
        if (xInt > yInt) return -1;
        if (yInt > xInt) return 1;
    }
    else
    {
        var compValue = x[i].CompareTo(y[i]);
        if (compValue != 0) return compValue;
    }
}
```

If the strings are integers (e.g. the `234` in `v1.234.56` and the `78` in `v1.78.0`), then it should compare 234 to 78 as integers and show that 234 comes after it.

(Or maybe I missed something.)

Reply:
I tested and it does not sort integers like that when you pass them as strings.  That said the string comparison looks like its only when something is not a number so I think it is OK.  My comment was just explaining why I crossed out my initial comment.
```
> list
List<string>(0) { }
> list.Add("234");
> list.Add("73");
> list.Sort()
> list
List<string>(2) { "234", "73" }
```